### PR TITLE
Feat: Maintain an active session until it expires

### PR DESCRIPTION
**Pull request by RichardIrala**
Pages affected:
* index
* inventory

Files modified:
* functions/login.js
* pages/index.js
* viewmodels/xerial_wallet.js

New file:
* helpers/session.js

Comment:
Includes login and logout in the UI.

### CODE REVIEW

File: functions/login.js [hunk 1]
```diff
@@ -39,4 +41,62 @@ const loginWithMetamask = async function ({ projectId }) {
   }
 };
 
-export default loginWithMetamask;
+const loadSession = async function () {
```
Comment (luislucena16) [inline functions/login.js:44]: why don't you put this loadSession in a try/catch all the if's and each one has a corresponding error handle?

File: functions/login.js [hunk 2]
```diff
@@ -39,4 +41,62 @@ const loginWithMetamask = async function ({ projectId }) {
...
+        },
+      }
+    );
+
+    const userResjson = await response.json();
+    
+    return { loguedWith, player: userResjson.user, sessionToken: tokens.access.token, tokens, wallets: userResjson.wallets }
+  } catch (error) {
+    console.log(error);
+    // throw new Error("Error to login.");
```
Comment (luislucena16) [inline functions/login.js]: delete this

File: pages/index.js [hunk 3]
```diff
@@ -113,14 +156,24 @@ export default function Home() {
...
             )}
           </div>
+          {!loguedWith && (
+            <button className="inventory__button" onClick={connectWallet}>
+              Connect with MetaMask
+            </button>
+          )}
+          {loguedWith === "metamask" && (
+            <button className="inventory__button" onClick={logoutAndClearUI}>
+              logout
```
Comment (luislucena16) [inline pages/index.js]: rename `Logout`

File: viewmodels/xerial_wallet.js [hunk 4]
```diff
@@ -82,24 +84,28 @@ export default function XerialWalletViewmodel(helpers) {
...
+      } else throw new Error("Auth player failed")
     } catch (error) {
       console.log(error);
-      throw new Error("Error to login.");
+      throw new Error("Error to login");
     }
   }
 
-  vm.loadSession = async function ({ tokens, loguedWith }) {
+  vm.loadSession = async function () {
```
Comment (luislucena16) [inline viewmodels/xerial_wallet.js:98]: maybe you can try the same as above with the try/catch?

**Review by luislucena16 — CHANGES_REQUESTED** [review]

File: functions/login.js (hunk 1 above)
Comment (RichardIrala) [reply functions/login.js:44]: I changed it. Thank you.

File: functions/login.js (hunk 2 above)
Comment (RichardIrala) [reply functions/login.js]: Done

File: pages/index.js (hunk 3 above)
Comment (RichardIrala) [reply pages/index.js]: Done

File: viewmodels/xerial_wallet.js (hunk 4 above)
Comment (RichardIrala) [reply viewmodels/xerial_wallet.js:98]: Done, yeah. I implement same solution with trycatch

**Comment by luislucena16** [comment]
approve!!

**Review by luislucena16 — APPROVED** [review]

**Pull request merged**